app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -421,9 +428,10 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         }
 
         if (!browserOverlay.isVisible && session.isYoutubeTV &&
-                event.keyCode == KeyEvent.KEYCODE_BACK) {
-            val escKeyEvent = KeyEvent(event.action, KeyEvent.KEYCODE_ESCAPE)
-            (activity as MainActivity).dispatchKeyEvent(escKeyEvent)
+                event.keyCode == KeyEvent.KEYCODE_BACK &&
+                !youtubeOverride) {
+            youtubeKeyEvents.add(event)
+            webView?.evalJS("youtubeBackHandler.youtubeBack(document.activeElement.parentElement.parentElement.id === 'guide-list');")
```

Comment:
Did you verify this is the best way to identify if we're at the top level in YouTube's back stack? I did this rather quickly.

Also, we should clarify what `document.active.parent.parent` actually means and why it works, preferably in code but in comments if that fails.